marathon/src/main/scala/io/buoyant/marathon/v2/AppIdNamer.scala

```diff
@@ -45,16 +45,23 @@ class AppIdNamer(
       appsActivity.map { apps =>
         Trace.recordBinary("marathon.path", path.show)
         val found = possibleIds.collectFirst {
-          case app if apps(app) =>
+          case app if apps(app) => {
             Trace.recordBinary("marathon.appId", app.show)
             val residual = path.drop(app.size)
             val id = prefix ++ app
             val addr = getAndMonitorAddr(app)
-            Name.Bound(addr, id, residual)
+            addr.map {
```

Comment:
This is a `Var[NameTree[Name]]` but I think it needs to be `Activity[NameTree[Name]]` to match with the flatMap (see comment above).

You can use `addr.map` to map Addr states into `Activity.State` and get a `Var[Activity.State[NameTree[Name]]]`.  Then you can use `Activity.apply` to turn that into an `Activity[NameTree[Name]]`.